app/models/robot/pick_data.rb

```diff
@@ -57,6 +63,91 @@ def generate_picking_data
     data_object
   end
 
+  # processes cherrypicking requests for a single batch and destination plate
+  # if there are more source plates than the maxiumum capacity for the robot, splits it out into multiple picks
+  # returns a hash of pick number (1-indexed) to data object containing info about source and destination plates
+  # basic hash structure:
+  # {
+  #   1 => {
+  #     'destination' => {},
+  #     'source' => {},
+  #     'time' => Time.zone.now,
+  #     'user' => user.login
+  #   },
+  #   2 => { etc. }
+  # }
+  # see pick_data_spec.rb for the more detailed structure
+  def generate_picking_data_hash
+    data_objects = {}
+    source_barcode_to_pick_number = {}
+
+    current_pick_size = lambda do
+      data_objects[data_objects.size]['source'].size
+    end
+
+    requests_for_destination_plate.find_each do |request|
+      target_plate = request.target_asset.plate
+      next unless target_plate.any_barcode_matching?(target_barcode)
+
+      source_barcode = request.asset.plate.machine_barcode
+
+      # find if barcode already encountered
+      pick_to_use = source_barcode_to_pick_number[source_barcode]
+
+      unless pick_to_use
+        # if no max beds, default to all in one pick
+        if @max_beds.nil?
+          pick_to_use = 1
```

Comment:
See comment above. A default value of Float::INFINITY for eliminate the need for this check, but might not be the right behaviour anyway.